src/main/daml/Daml/Finance/Claims/Lifecycle/Rule.daml

```diff
@@ -81,3 +80,17 @@ template Rule
               toInterface @Disclosure.I claimInstrument
           pure (producedInstrument, [effectCid])
         -- BOND_PROCESS_CLOCK_UPDATE_LIFECYCLE_END
+
+-- | HIDE
+-- Check if an instrument for the new key already exists, otherwise create it.
+tryCreateNewInstrument : Party -> Time -> InstrumentKey-> InstrumentKey -> Update ()
+tryCreateNewInstrument actor eventTime oldKey newKey =
```

Comment:
Do we need the key check? I think it wasn't there in the original version of the rule and I am trying to understand why we would need it now

Reply:
I opened this separate issue to ensure unique versions: https://github.com/digital-asset/daml-finance/issues/752